src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -794,21 +797,17 @@ private uint GetParameterValEscape(ParameterSymbol parameter)
 
         private uint GetParameterRefEscape(ParameterSymbol parameter)
         {
-            if (UseUpdatedEscapeRules)
```

Comment:
I suspect that more places where this check is used can be simplified. If the ParameterSymbol APIs behave appropriately depending on which escape rules are in use, then doing these checks separately could be redundant.